### PR TITLE
crimson/doc: extend and update building guidelines

### PR DESCRIPTION
**Pull request by rzarzynski**
0. Extend the instruction with the info on how to clone the repo.
1. Use the `do_cmake.sh` wrapper, which takes care about submodules, instead of invoking CMake manually.
2. Switch from GNU Make to Ninja.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: doc/dev/crimson/crimson.rst [hunk 1]
```diff
@@ -27,7 +37,7 @@ the alien threads. So, to use alienstore / bluestore backend, you might want to
 pass ``-DSeastar_CXX_FLAGS=-DSEASTAR_DEFAULT_ALLOCATOR`` to ``cmake`` when
 configuring this project to use the libc allocator, like::
 
-  $ cmake -DWITH_SEASTAR=ON -DSeastar_CXX_FLAGS=-DSEASTAR_DEFAULT_ALLOCATOR ..
+  $ ./do_cmake.sh -DWITH_SEASTAR=ON -DSeastar_CXX_FLAGS=-DSEASTAR_DEFAULT_ALLOCATOR ..
```
Comment (tchaikov) [inline doc/dev/crimson/crimson.rst:40]: this change conflicts with #44411

File: doc/dev/crimson/crimson.rst [hunk 2]
```diff
@@ -10,14 +10,24 @@ be backward compatible with classic OSD.
...
 ================
 
 Crimson is not enabled by default. To enable it::
 
   $ WITH_SEASTAR=true ./install-deps.sh
-  $ mkdir build && cd build
-  $ cmake -DWITH_SEASTAR=ON ..
+  $ ./do_cmake.sh -DWITH_SEASTAR=ON
+  $ cd build
+  $ ninja
```
Comment (tchaikov) [inline doc/dev/crimson/crimson.rst:29]: personally, i am not in favor of replicating the document which also applies to Ceph in general.

File: doc/dev/crimson/crimson.rst (hunk 1 above)
Comment (rzarzynski) [reply doc/dev/crimson/crimson.rst:40]: Will rebase.

File: doc/dev/crimson/crimson.rst (hunk 2 above)
Comment (rzarzynski) [reply doc/dev/crimson/crimson.rst:29]: I wasn't too. I changed my mind when I had to provide a step-by-step instruction to non-developers. Interleaving two docs brought huge amount of confusion :-(.

**Pull request closed without merging**